File: backend/src/domain/entities/Fund.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('funds')
export class Fund {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'organization_id' })
  organizationId: number;

  @Column({ name: 'manager_member_id' })
  managerMemberId: number;

  @Column({ type: 'varchar', nullable: true })
  name: string;

  @Column({ type: 'text', nullable: true })
  description: string;

  @Column({
    type: 'decimal',
    name: 'target_amount',
    nullable: true,
    precision: 15,
    scale: 2,
  })
  targetAmount: number;

  @Column({
    type: 'decimal',
    name: 'current_amount',
    nullable: true,
    precision: 15,
    scale: 2,
  })
  currentAmount: number;

  @Column({ type: 'varchar', nullable: true })
  status: string;
}

File: backend/src/main/java/com/service/backend/domain/entity/Fund.java
package com.service.backend.domain.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("funds")
public class Fund {

    @Id
    private Long id;

    @Column("organization_id")
    private Long organizationId;

    @Column("manager_member_id")
    private Long managerMemberId;

    private String name;

    private String description;

    @Column("target_amount")
    private BigDecimal targetAmount;

    @Column("current_amount")
    private BigDecimal currentAmount;

    private String status;
}
